src/app.js

```diff
@@ -1,55 +1,152 @@
-const bodyParser = require('body-parser');
-const express = require('express');
-const GitHubApi = require('github');
-const nacl = require('tweetnacl');
-nacl.util = require('tweetnacl-util');
-
-const username = 'yourusername';  // TODO: your GitHub username here
-const github = new GitHubApi({ debug: true });
+const bodyParser = require("body-parser");
+const express = require("express");
+const GitHubApi = require("github");
+const nacl = require("tweetnacl");
+nacl.util = require("tweetnacl-util");
+
+require("dotenv").config();
+
+const username = "Lambdarines"; // TODO: your GitHub username here
+const github = new GitHubApi({
+  debug: true
+});
 const server = express();
+const token = process.env.GITHUB_TOKEN;
+let client_id = "";
+// console.log('token', token);
+
+let myPublicKey;
+
+
+server.use(bodyParser.json());
 
 // Generate an access token: https://github.com/settings/tokens
 // Set it to be able to create gists
 github.authenticate({
-  type: 'oauth',
-  token: process.env.GITHUB_TOKEN
+  type: "oauth",
+  token
 });
 
 // Set up the encryption - use process.env.SECRET_KEY if it exists
 // TODO either use or generate a new 32 byte key
+// const secretKey = Uint8Array.from(process.env.SECRET_KEY);
+// const nonce = process.env.NONCE;
 
-server.get('/', (req, res) => {
+server.get("/", (req, res) => {
   // TODO Return a response that documents the other routes/operations available
+  github.users.getForUser({
+    username
+  }).then(response => {
+    res.json(response.data);
+    console.log("client id", response.data.id);
+    client_id = response.data.id;
+    // console.log(response.data);
+  });
 });
 
-server.get('/gists', (req, res) => {
+server.get("/gists", (req, res) => {
   // TODO Retrieve a list of all gists for the currently authed user
+  github.gists.getForUser({
+    username
+  }).then(response => {
+    res.json(response.data);
+  });
 });
 
-server.get('/key', (req, res) => {
+server.get("/key", (req, res) => {
   // TODO Return the secret key used for encryption of secret gists
+  res.send(nacl.util.encodeBase64(key));
 });
 
-server.get('/secretgist/:id', (req, res) => {
-  // TODO Retrieve and decrypt the secret gist corresponding to the given ID
+server.get("/secretgist/:id", (req, res) => {
+  try {
```

Comment:
this might be because you all worked in a group, but try to stay consistent with the coding  style. In this function you are using a try catch block while in all the previous ones this wasn't the case.

Reply:
Just want to strongly :+1: this, one of the biggest challenges of multi-people projects is consistency. Companies often have internal code style guides for this, for public projects just try to agree on these sorts of things up front.